Update README.md

Rosa-ccyyds opened the pull request:
I have updated the initial version of README

Rosa-ccyyds committed:
Update README.md

SusanLin0426 reviewed (changes requested):
It might be good to also add List, Update, and Delete endpoints so that we cover the full CRUD functionality

Rosa-ccyyds committed:
Update README.md

Specifically, it replaces the term “Retrieve” with “Read” and adds documentation for the List, Update, and Delete endpoints to complete the CRUD functionality.

SusanLin0426 commented on README.md:
````diff
@@ -1,69 +1,143 @@
+### Delete a Recommendation
+```
+DELETE /recommendations/<id>
+→ 204 No Content | 404 Not Found
````
Think we can keep only 204, since 404 won't occur

SusanLin0426 reviewed (commented):
Delete endpoint only have 204, think we can keep only 204?

SusanLin0426 reviewed (commented):
Think we don't need to include " Roadmap" section

Rosa-ccyyds committed:
Update README.md
